### PR TITLE
OCPBUGS-62828: MCO degraded when an empty pull-secret is configured

### PR DESCRIPTION
**Pull request by dkhater-redhat**
**- What I did**
  Fixed a bug where MCO becomes degraded when an empty pull-secret is configured in the cluster. Modified
  `getImageRegistryPullSecrets()` in `pkg/operator/sync.go` to gracefully handle empty pull-secrets by
  checking if the pull-secret data is empty or contains only `{}` before attempting to convert and merge
  it. When an empty pull-secret is detected, the operator now skips the merge operation instead of
  erroring out. Also added test coverage in `pkg/secrets/secrets_test.go` for empty JSON object and null
  literal edge cases.
**- How to verify it**
    1. Configure an empty pull-secret in the cluster 

$ oc -n openshift-config set data secret/pull-secret '.dockerconfigjson={}'

and verify MCO is not degraded. 

**- Description for the changelog**

### CODE REVIEW

File: pkg/secrets/secrets_test.go [hunk 1]
```diff
@@ -179,6 +179,16 @@ type: kubernetes.io/dockercfg`
 			bytes:    []byte(`{"auths":{}}`),
 			expected: &DockerConfigJSON{Auths: DockerConfig{}},
 		},
+		{
```
Comment (cheesesashimi) [inline pkg/secrets/secrets_test.go:182]: praise: Thanks for adding these test cases!

**Comment by sergiordlr** [comment]
Verified using IPI on AWS

Test case OCP-57595 has been modified to check the machine-config CO, so that this issue doesn't escape again.

Executed test cases:

- "[sig-mco] MCO Author:sregidor-NonPreRelease-Longduration-Medium-57595-[OnCLayer] Use empty pull-secret[Disruptive] [Serial]"
- "[sig-mco] MCO Author:mhanss-NonPreRelease-Longduration-High-42680-[P2] change pull secret in the openshift-config namespace [Serial]"

A MC deploying a file was properly deployed too.

No issues found.

/label qe-approved
/verified by @sergiordlr

**Comment by umohnani8** [comment]
Great work! LGTM
